systemFlake.nix

```diff
@@ -9,11 +9,14 @@
 , channelsConfig ? { }
 , sharedOverlays ? [ ]
 
-, hostDefaults ? { }
 , nixosProfiles ? { } # will be deprecated soon, use hosts, instead.
 , hosts ? nixosProfiles
-, sharedExtraArgs ? { }
-, sharedModules ? [ ]
+, sharedExtraArgs ? { } # deprecate soon, prefer hostDefaults
+, sharedModules ? [ ] # deprecate soon, prefer hostDefaults
+, hostDefaults ? {
+    modules = sharedModules;
+    extraArgs = sharedExtraArgs;
+  }
```

Comment:
So if you set hostDefaults, you can't use sharedModules. I think this is good deprecation policy.